src/Execution/lbsRuntime.py

```diff
@@ -264,25 +264,24 @@ def transfer_stage(self, transfer_criterion, deterministic_transfer):
                         c = transfer_criterion.compute(o, p_src, p)
                         if c < 0.:
                             n_rejects += 1
-                        if c >= c_value:
+                        elif c > c_value:
                             c_value = c
                             p_dst = p
-
+                    
                     # Move to next object if no transfer was possible
                     if not p_dst:
                         continue
 
                 else:
                     # Compute transfer CMF given information known to source
-                    p_cmf = p_src.compute_transfer_cmf()
+                    p_cmf = p_src.compute_transfer_cmf(
+                        transfer_criterion, o, targets, True)
```

Comment:
Have you tried probabilistic with strict=True for pure loads? In TemperedLB, when I had tried omitting anything from the cmf that would later fail the criterion, it seemed to have trouble converging on a good solution. I'm wondering if you've seen the same thing in LBAF and if it might impact total work too.

Reply:
Yes same thing Nicole!
If I turn the strict flag to `True` it's not converging as well as when it's `False`.
I was a bit puzzled by that but am now reassured as you're describing a similar pattern with TemperedLB. Thanks!

Reply:
Does that still happen with `actual_destination_load: True`?

Reply:
Yes!

Reply:
Interesting. My initial thought had been that it was related to inadvertently overloading the most underloaded processors by not knowing what other processors are doing, but it sounds like you've ruled that out. Have you experimented with strict=True under different object orderings?

Reply:
Not yet because I wanted to explore that as part of #121 but it's likely these 2 issues (#122 and #121) are not entirely disjoint. 

What I find encouraging about what we are finding in the diagrams above, is that the criterion is working as expected, enforcing monotonicity (decreasing) of the objective function (total work). It's not converging to the absolute minimum due to the atomic transfers -- but at least it's not worsening the situation at any iteration.

Reply:
Yes, that's definitely good. I'm very interested in where all this goes.